youtube_dl/extractor/spankwire.py

```diff
@@ -54,61 +49,48 @@ def _real_extract(self, url):
 
-        title = self._html_search_regex(
+        title = video_data.get('title') or self._html_search_regex(
             r'<h1>([^<]+)', webpage, 'title')
-        description = self._html_search_regex(
-            r'(?s)<div\s+id="descriptionContent">(.+?)</div>',
-            webpage, 'description', fatal=False)
-        thumbnail = self._html_search_regex(
-            r'playerData\.screenShot\s*=\s*["\']([^"\']+)["\']',
-            webpage, 'thumbnail', fatal=False)
+        thumbnail = video_data.get('poster')
+        description = video_data.get('description')
 
-        uploader = self._html_search_regex(
-            r'by:\s*<a [^>]*>(.+?)</a>',
-            webpage, 'uploader', fatal=False)
+        uploader = self._search_regex(
+            r'<a[^>]+class="uploaded__by"[^>]*>(.+?)</a>',
+            webpage, 'uploader', flags=re.DOTALL, fatal=False)
         uploader_id = self._html_search_regex(
-            r'by:\s*<a href="/(?:user/viewProfile|Profile\.aspx)\?.*?UserId=(\d+).*?"',
+            r'by\s*<a href="/(?:user/viewProfile|Profile\.aspx)\?.*?UserId=(\d+).*?"',
             webpage, 'uploader id', fatal=False)
         upload_date = unified_strdate(self._html_search_regex(
-            r'</a> on (.+?) at \d+:\d+',
+            r'</span>(.+?) at \d+:\d+ (?:AM|PM) by',
             webpage, 'upload date', fatal=False))
 
-        view_count = str_to_int(self._html_search_regex(
-            r'<div id="viewsCounter"><span>([\d,\.]+)</span> views</div>',
-            webpage, 'view count', fatal=False))
-        comment_count = str_to_int(self._html_search_regex(
-            r'<span\s+id="spCommentCount"[^>]*>([\d,\.]+)</span>',
-            webpage, 'comment count', fatal=False))
+        view_count = int_or_none(video_data.get('viewed'))
+        comment_count = int_or_none(video_data.get('comments'))
+        duration = int_or_none(video_data.get('duration'))
+
+        def extract_list(arr):
+            names = []
+            for i in arr:
+                names.append(i.get('name'))
+            return names
 
-        videos = re.findall(
-            r'playerData\.cdnPath([0-9]{3,})\s*=\s*(?:encodeURIComponent\()?["\']([^"\']+)["\']', webpage)
-        heights = [int(video[0]) for video in videos]
-        video_urls = list(map(compat_urllib_parse_unquote, [video[1] for video in videos]))
-        if webpage.find(r'flashvars\.encrypted = "true"') != -1:
-            password = self._search_regex(
-                r'flashvars\.video_title = "([^"]+)',
-                webpage, 'password').replace('+', ' ')
-            video_urls = list(map(
-                lambda s: aes_decrypt_text(s, password, 32).decode('utf-8'),
-                video_urls))
+        categories = extract_list(video_data.get('categories'))
+        tags = extract_list(video_data.get('tags'))
 
+        videos = video_data.get('videos').items()
+        videos.sort()
```

Comment:
1. Does not work in python 3.
2. Sorting is invalid. Sorting must be done with _sort_formats.